lib/vagrant-service-manager/plugin.rb

```diff
@@ -18,12 +18,14 @@ class Plugin < Vagrant.plugin('2')
         Config
       end
 
-      action_hook(:servicemanager, :machine_action_up) do |hook|
-        hook.before(VagrantPlugins::ProviderVirtualBox::Action::Network, setup_network)
-        hook.after(Vagrant::Action::Builtin::SyncedFolders, Service)
-        hook.after Vagrant::Action::Builtin::Provision, final_actions
+      service_manager_hooks = lambda do |hook|
```

Comment:
Did you test this with other providers than VirtualBox? Does it work with libvirt? Did you try on Windows? My experience is that these hooks are very sensitive.